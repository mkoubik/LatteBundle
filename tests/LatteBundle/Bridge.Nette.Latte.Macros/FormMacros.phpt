<?php

require_once __DIR__ . '/../bootstrap.php';

use Symfony\Component\HttpKernel\Bundle\Bundle;

class TestingBundle extends Bundle
{
}

require_once __DIR__ . '/../../integration/kernel.php';

$kernel = new TestKernel('test', false);
$request = Symfony\Component\HttpFoundation\Request::create('http://example.com/home');
$kernel->handle($request);
$container = $kernel->getContainer();
$container->set('request', $request);
$container->enterScope('request');
$engine = $container->get('templating.engine.latte');


$factory = Symfony\Component\Form\Forms::createFormFactory();


$form1 = $factory->createBuilder()
    ->add('create', 'submit', array(
        'label' => 'Create',
        'attr' => array('class' => 'button'),
    ))
    ->getForm();

$html = $engine->render('TestingBundle:test:formMacros.html.latte', array(
    'form1' => $form1->createView(),
));
Assert::matchFile(__DIR__ . '/expected/formMacros.html', $html);
