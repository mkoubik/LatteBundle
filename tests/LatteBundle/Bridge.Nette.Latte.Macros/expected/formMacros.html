<form method="post" action="" class="inline-form">
<button type="submit"
    id="form_create"
    name="form[create]" class="button">Create</button>
</form>

<input type="file"
    id="form_image"
    name="form[image]"
 accept="image/*" value="">

<input type="email"
    id="form_email"
    name="form[email]"
 required="required" maxlength="255" value="" class="form-control">
